docs/using-connect/integration.mdx

```diff
@@ -0,0 +1,374 @@
+	oraclePreBlockHandler := oraclepreblock.NewOraclePreBlockHandler(
+		app.Logger(),
+		aggregatorFn,
+		app.OracleKeeper,
+		oracleMetrics,
+		currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+		veCodec,
+		ecCodec,
+	)
+
+	app.SetPreBlocker(oraclePreBlockHandler.WrappedPreBlocker(app.ModuleManager))
+}
+```
+
+Next, configure the vote extensions using the vote extension codec, extended commit codec, and aggregator function from the previous step.
+
+```go oracle.go
+package app
+
+import (
+	"time"
+
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+
+	"github.com/skip-mev/slinky/abci/ve"
+	"github.com/skip-mev/slinky/abci/strategies/currencypair"
+	"github.com/skip-mev/slinky/abci/strategies/aggregator"
+)
+
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {
+
+	// ... snip ...
+
+	// Create the vote extensions handler that will be used to extend and verify
+	// vote extensions (i.e. oracle data).
+	voteExtensionsHandler := ve.NewVoteExtensionHandler(
+		app.Logger(),
+		oracleClient,
+		time.Second, // timeout
+		currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+		veCodec,
+		aggregator.NewOraclePriceApplier(
+			aggregator.NewDefaultVoteAggregator(
+				app.Logger(),
+				aggregatorFn,
+				// we need a separate price strategy here, so that we can optimistically apply the latest prices
+				// and extend our vote based on these prices
+				currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+			),
+			app.OracleKeeper,
+			veCodec,
+			ecCodec,
+			app.Logger(),
+		),
+		oracleMetrics,
+	)
+	app.SetExtendVoteHandler(voteExtensionsHandler.ExtendVoteHandler())
+	app.SetVerifyVoteExtensionHandler(voteExtensionsHandler.VerifyVoteExtensionHandler())
+}
+```
+
+Finally, call these methods back in `app.go`, directly after setting the `x/marketmap` hooks.
+
+```go app.go
+	app.MarketMapKeeper.SetHooks(app.OracleKeeper.Hooks())
+
+	// oracle initialization
+	client, metrics, err := app.initializeOracle(appOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize oracle client and metrics: %w", err)
+	}
+
+	app.initializeABCIExtensions(client, metrics)
+```
+
+## Starting the Application
+
+In order for the application to use Connect properly, the following is required:
+
+- Set the consensus parameters to enable vote extensions
+- Initialize `x/marketmap` with initial markets
+
+The following code can be executed within an upgrade handler for chains that are already live or in `InitChainer` for new chains:
+
+```go
+package app
+
+import (
+	"slices"
+
+    tmtypes "github.com/cometbft/cometbft/proto/tendermint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	consensustypes "github.com/cosmos/cosmos-sdk/x/consensus/types"
+	"github.com/skip-mev/slinky/cmd/constants/marketmaps"
+)
+
+func (app *App) setupMarkets(ctx sdk.Context) error {
```

Comment:
I think we could also export a function like this from our codebase that just takes in a few parameters:

1. enable height
2. desired markets
3. desired market map params

Reply:
opened https://linear.app/skip/issue/BLO-1577/expose-function-to-ease-integration-of-market-setup-ve-enabled-height